warn-stranded.lua

```diff
@@ -333,8 +333,9 @@ local function getStrandedUnits()
                 or getWalkGroup(xyz2pos(unitPos.x+1, unitPos.y+1, unitPos.z))
                 or 0
         end
-
-        if unitIgnored(unit) then
+		
+        -- Ignore units who are gathering plants to avoid errors with stepladders
+        if unitIgnored(unit) or unit.job.current_job.job_type == df.job_type.GatherPlants then
```

Comment:
are you guaranteed that `current_job` is non-nil at this point?

Reply:
Good call. I think in my test fortress everyone was super busy so that didn't get caught. I'll check into that and update it as needed.